Create updated tests for APG design pattern example: checkbox

zcorpan opened the pull request:
Part of #443

---

This is currently a work-in-progress. The changes in `commands.csv` at least are not complete.

zcorpan commented:
For some reason `npm run create-all-tests` doesn't create the tests for `checkbox`. I've probably made a mistake somewhere, but there's no useful error message. I haven't yet figured out what's wrong.

zcorpan commented:
OK, I think everything works now! Ready for review.

jscholes commented:
@zcorpan 

#### Example

1. The `reference` directory contains assets relating to the tri-state checkbox (checkbox2) as well as the one under test.
2. The default JS/CSS references haven't been removed from the example (e.g. the W3C stylesheet, references to JS assets from the APG repo that we don't use).
3. The example has been modified to the degree that the heading structure now skips levels (`<h1>` followed by `<h3>`).  Usually we don't cut this much content out.

For more info, see https://github.com/w3c/aria-at/issues/394.  Some of these issues seem to have already been present in the checkbox plan, but we should probably get them fixed up if we're submitting updates to it.

#### commands.csv

The JAWS commands still include smart navigation commands which are no longer in scope.

#### tests.csv

1. The navigation tests don't include any state in their names.  E.g. test 1 and test 9 have the same title ("Navigate forwards to checkbox in reading mode").
2. The inclusion of an article in test titles is inconsistent.  E.g. "Navigate forwards to checkbox" versus "Operate a checkbox" (no "a" in the first one).
3. The "operate a checkbox" tests aren't separated based on the state transition.  I.e. we need tests to check a checkbox, and separate ones to uncheck it.

I have some additional thoughts about the group-related tests, but I'll revisit those.

#### Test Preview

The index page is working, but the [review page for checkbox](https://raw.githack.com/w3c/aria-at/bocoup/update-checkbox-tests/build/review/checkbox.html) seems completely broken.  For example I see:

* Test 1: Navigate forwards to checkbox in reading mode
* Test 2: Navigate to an unchecked checkbox in reading mode

The second of those isn't even present in `tests.csv`?  So I have no idea where it's coming from.  The same applies all the way down, with old tests interleaved with the new.

zcorpan commented:
I've addressed the comments on the source files.

I don't know what was up with the test preview.

@jscholes 

> I have some additional thoughts about the group-related tests, but I'll revisit those.

What about them? :)
